lib/common/fs-operation.js

```diff
@@ -0,0 +1,251 @@
+        var listMount = 'df';
+
+        return this._PromisedChildProcess(listMount);
+    };
+
+    FsOperation.prototype.unmountAllIso = function unmountIso() {
+        var self = this;
+        return this.getAllMounts()
+            .then(function(cmdOutput){
+                var mountLines = cmdOutput.stdout.split('\n');
+                var mountDirectories = [];
+
+                _.forEach(mountLines, function(mountLine){
+                    if(mountLine.indexOf(process.cwd()) >= 0) {
+                        var mountDirectory = mountLine.split(/\s+/)[5];
+                        mountDirectories.push(mountDirectory);
+                    }
+                });
+
+                return Promise.each(mountDirectories, function(mountDirectory){
+                    return self.unmountIso(mountDirectory);
+                });
+            });
+    };
+
+    FsOperation.prototype.createSymbolLink = function createSymbolLink(
+        source,
+        dest) {
+            logger.info('Creating symbol link to ' + dest);
+
+            // create parent direct if it's not exists
+            this.prepareWritableDir(path.join(dest, '../'));
+
+            var ln = 'ln -sf ' + source + ' ' + dest;
+
+            return this._PromisedChildProcess(ln);
+    };
+
+    FsOperation.prototype.removeDir = function removeDir(
+        link
+    ) {
+        try {
+            logger.info('Removing folder ' + link);
+            fs.rmdirSync(link);
+        } catch (err) {
+            // non-critical error, continue anyway
+            logger.info(JSON.stringify(err.message));
+        }
+    };
+
+    FsOperation.prototype.removeParentDir = function removeParentDir(
+        link
+    ) {
+        try {
+            var parent = path.join(link, '../');
+
+            if(_.isEmpty(fs.readdirSync(parent))){
+                logger.info('Removing folder ' + parent);
+
+                fs.rmdirSync(parent);
+            }
+        } catch (err) {
+            // non-critical error, continue anyway
+            logger.info(JSON.stringify(err.message));
+        }
+    };
+
+    FsOperation.prototype.removeDirAndEmptyParent= function removeDirAndEmptyParent(
+        link
+    ) {
+        this.removeDir(link);
+        this.removeParentDir(link);
+    };
+
+    FsOperation.prototype.removeFile = function removeFile(
+        link
+    ) {
+        return Promise.try(function(){
+            logger.info('Removing file ' + link);
+            fs.unlinkSync(link);
+        });
+    };
+
+    FsOperation.prototype.getFileStat = function getFileStat(
+        link
+    ) {
+        try {
+            var stat = fs.statSync(link);
+            return stat;
+        } catch (err) {
+            return new Error('file not found');
+        }
+    };
+
+    FsOperation.prototype.getFileSize = function getFileSize(
+        link
+    ) {
+        var stat = this.getFileStat;
+
+        return stat['size'];
+    };
+
+    return new FsOperation();
+}
```

Comment:
Is there a better way to differentiate these two types of functions, async and sync, in this module? Like use “Sync” or “Async” keyword in the name of each function.